processing/src/main/java/org/apache/carbondata/processing/datatypes/GenericDataType.java

```diff
@@ -80,7 +82,13 @@ void writeByteArray(T input, DataOutputStream dataOutputStream)
    * @throws KeyGenException
    */
   void parseAndBitPack(ByteBuffer byteArrayInput, DataOutputStream dataOutputStream,
-      KeyGenerator[] generator) throws IOException, KeyGenException;
+      KeyGenerator[] generator)
+      throws IOException, KeyGenException;
+
+
+  int parseComplexValue(ByteBuffer byteArrayInput, DataOutputStream dataOutputStream,
+      KeyGenerator[] generator, Boolean[][] complexDictionaryIndentification, int startOffset)
+      throws IOException, KeyGenException;
```

Comment:
Remove `complexDictionaryIndentification` and `startOffset` if not used

Reply:
complexDictionaryIndentification is removed but StartOffset is used.